hawkbit-dmf/hawkbit-dmf-amqp/src/main/java/org/eclipse/hawkbit/amqp/InvalidTargetOperationsExceptionHandler.java

```diff
@@ -0,0 +1,31 @@
+/**
+ * Copyright (c) 2021 Bosch.IO GmbH and others.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.hawkbit.amqp;
+
+import org.eclipse.hawkbit.exception.ConditionalErrorHandler;
+import org.eclipse.hawkbit.exception.ErrorHandlerChain;
+import org.eclipse.hawkbit.repository.exception.EntityNotFoundException;
+import org.eclipse.hawkbit.repository.exception.InvalidTargetAttributeException;
+import org.springframework.amqp.AmqpRejectAndDontRequeueException;
+
+/**
+ * An error handler for all invalid target operations resulting from AMQP.
+ */
+public class InvalidTargetOperationsExceptionHandler implements ConditionalErrorHandler<Throwable> {
+
+    @Override
+    public void doHandle(final Throwable t, final ErrorHandlerChain<Throwable> chain) {
+        Throwable cause = t.getCause();
```

Comment:
pease verify if taking always the first parent/cause is sufficient here or should we recursively parse the stack, please provide comment, thanks!

Reply:
yes for the target operation, first parent is sufficient as the children cause also contain the same cause